kubectl/kubectl-deploy.yaml

```diff
@@ -5,14 +5,60 @@ metadata:
   namespace: default
 spec:
   params:
+  - name: action
+    description: Action on the resource
+  - name: merge-strategy
+    default: strategic
+    description: Merge strategy when using action  patch
   - name: manifest
     description: Content of the resource to deploy
+  - name: success-condition
+    default: ""
+    description: A label selector express to decide if the action on resource is success.
+  - name: failure-condition
+    default: ""
+    description: A label selector express to decide if the action on resource is failure.
+  - name: output
+    default: ""
+    description: An express to retrieval data from resource.
+  - name: set-ownerreference
+    default: "false"
+    description: Enable set owner reference for created resource.
   - name: image
-    default: gcr.io/cloud-builders/kubectl # it is huge
+    default: index.docker.io/aipipeline/kubeclient:v0.0.2 # it is huge
```

Comment:
What is the difference between this image and `gcr.io/cloud-builders/kubectl`? Just had a query :sweat_smile:

Reply:
@vinamra28 The `gcr.io/cloud-builders/kubectl` only supply cmd `kubectl`, so we build another one based on it to support more complex features: `success-condition/failure-conditio`, `set-ownerreference` .etc.

Reply:
Okay. Thanks